### PR TITLE
11665 XCN mappings

### PR DESCRIPTION
**Pull request by jack-h-wang**
This PR adds the XCN <-> Practitioner mapping.

The XCN <-> PractitionerRole mapping will be inlined when implementing the segments that use it.

Test Steps:
1. Review and confirm test files and results are as expected.

## Changes
- Added XCN <-> Practitioner mapping.

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] Added tests?

## Linked Issues
- #11665

### CODE REVIEW

File: prime-router/metadata/fhir_mapping/hl7/datatypes/XCN/Practitioner.yml [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+          valueOf: "GeneralUtils.dateTimeWithZoneId(periodStart,ZONEID)"
+          expressionType: JEXL
+          vars:
+            periodStart: XCN.19
+        end:
+          valueOf: "GeneralUtils.dateTimeWithZoneId(periodEnd,ZONEID)"
+          expressionType: JEXL
+          vars:
+            periodEnd: XCN.20
+    extension_AssemblyOrder:
```
Comment (jack-h-wang) [inline prime-router/metadata/fhir_mapping/hl7/datatypes/XCN/Practitioner.yml:161]: The mapping inventory placed this extension within `name.family`, but per the extension definition, `name` is the correct location.

File: prime-router/metadata/hl7_mapping/resources/Practitioner/XCN.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    resource: '%resource.extension(%`rsext-xcn22-assigning-jurisdiction`).value'
+    constants:
+      cweFieldPath: '%{hl7XCNField}-22'
+    schema: ../../datatypes/codeableConcept/CWE
+
+  - name: xcn-assigning-agency-or-department
+    resource: '%resource.extension(%`rsext-xcn23-assigning-agency-or-department`).value'
+    constants:
+      cweFieldPath: '%{hl7XCNField}-23'
+    schema: ../../datatypes/codeableConcept/CWE
```
Comment (jack-h-wang) [inline prime-router/metadata/hl7_mapping/resources/Practitioner/XCN.yml:111]: `XCN.24` and `XCN.25` aren't present in the HL7 v2.5.1 or NIST specs.

File: prime-router/metadata/fhir_mapping/hl7/datatypes/XCN/Practitioner.yml [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+    # the values do not get correctly loaded and will not be mapped over correctly using a spec.
+    period_1:
+      vars:
+        xcn19: String, XCN.19
+        xcn20: String, XCN.20
+        start: String, XCN.17.1
+        end: String, XCN.17.2
+      condition: $xcn19 NULL && $xcn20 NULL
+      expressionType: resource
+      valueOf: datatypes/DR/Period
```
Comment (mkalish) [inline prime-router/metadata/fhir_mapping/hl7/datatypes/XCN/Practitioner.yml:142]: 🤦🏻 Typo in the inventory where the Data Type is missing

File: prime-router/src/testIntegration/kotlin/datatests/mappinginventory/xcn/XCNtoPractitionerTests.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package gov.cdc.prime.router.datatests.mappinginventory.xcn
+
+import gov.cdc.prime.router.datatests.mappinginventory.verifyHL7ToFHIRToHL7Mapping
+import org.junit.jupiter.api.Test
+
+class XCNtoPractitionerTests {
+    @Test
+    fun `test translate to HL7 to FHIR to HL7`() {
```
Comment (mkalish) [inline prime-router/src/testIntegration/kotlin/datatests/mappinginventory/xcn/XCNtoPractitionerTests.kt:8]: Nit: could we split this up into separate tests?  I think it makes it slightly better as documentation of how we expect things to work

File: prime-router/src/testIntegration/kotlin/datatests/mappinginventory/xcn/XCNtoPractitionerTests.kt [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        assert(verifyHL7ToFHIRToHL7Mapping("xcn/xcn-to-practitioner-xcn19-20-populated-xcn17-empty").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("xcn/xcn-to-practitioner-xcn17-populated-xcn19-20-empty").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("xcn/xcn-to-practitioner-uuid-assigning-authority").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("xcn/xcn-to-practitioner-uuid-assigning-authority-no-namespace").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("xcn/xcn-to-practitioner-iso-assigning-authority-no-namespace").passed)
```
Comment (mkalish) [inline prime-router/src/testIntegration/kotlin/datatests/mappinginventory/xcn/XCNtoPractitionerTests.kt:13]: Can we add test for a different value for HD.3 like DNS or CLIA to verify that the organization gets created?

**Review by mkalish — APPROVED** [review]
Nice work!  A couple of small things to clean up, but otherwise good to go

File: prime-router/src/testIntegration/resources/datatests/mappinginventory/xcn/xcn-to-practitioner-iso-assigning-authority-no-namespace.hl7 [hunk 6]
```diff
@@ -0,0 +1,2 @@
+MSH|^~\&#|||||20230501102531-0400||ORU^R01^ORU_R01|3003786103_4988249_33033||2.5.1
+OBR|1|||68991-9^Epidemiologically Important Information^LN^^^^^^^^^^^2.16.840.1.113883.6.1||||||1^BEETHOVEN&VAN&BEETHOVEN&VAL&ROGER^LUDWIG^B^2ND^DR^MD^SRC^&AssigningSystem&ISO^B^A^NPI^DL^^A^NameContext^^G^20220501102531-0400^20230501102531-0400^MD^AssignJ^AssignA|||||||||||||||F
```
[the remark this answers is not in the document]
Comment (jack-h-wang) [reply prime-router/src/testIntegration/resources/datatests/mappinginventory/xcn/xcn-to-practitioner-iso-assigning-authority-no-namespace.hl7:2]: I tried removing it and while the generated FHIR seemed OK, the FHIR to HL7 output dropped the `OBR` segment entirely. I think we'll need to revisit this when we handle mapping that segment.

**Pull request merged**